Guides/TLS.md

```diff
@@ -0,0 +1,78 @@
+# TLS/SSL and MongoSwift
+
+MongoSwift supports connecting to MongoDB over TLS/SSL. This guide covers the installation requirements and configuration options for enabling this functionality in MongoSwift. See the [server documentation](https://docs.mongodb.com/manual/tutorial/configure-ssl/) to configure MongoDB to use TLS/SSL.
+
+## Dependencies
+
+MongoSwift relies on the the TLS/SSL library installed on your system for making secure connections to the database. 
+ - On macOS, the driver depends on SecureTransport, the native TLS library for macOS, so no additional installation is required.
+ - On Linux, the driver depends on OpenSSL, which is usually bundled with your OS but may require specific installation. The driver also supports LibreSSL through the use of OpenSSL compatibility checks.
+ 
+### Ensuring TLS 1.1+
+
+Industry best practices recommend, and some regulations require, the use of TLS 1.1 or newer. Though no application changes are required for MongoSwift to make use of the newest protocols, some operating systems or versions may not provide a TLS library version new enough to support them.
+
+#### ...on Linux
+
+Users of Linux or other non-macOS Unix can check their OpenSSL version like this:
+```
+$ openssl version
+```
+If the version number is less than 1.0.1, support for TLS 1.1 or newer is not available. Contact your operating system vendor for a solution, upgrade to a newer distribution, or manually upgrade your installation of OpenSSL.
+
+#### ...on macOS
+
+macOS 10.13 (High Sierra) and newer support TLS 1.1+.
+
+
+## Basic Configuration
+
+To connect to MongoDB over TLS/SSL, simply specify `tls: true` in the `ClientOptions` passed to a `MongoClient`'s initializer:
+```swift
+let client = try MongoClient("mongodb://example.com", using: elg, options: ClientOptions(tls: true))
+```
+
+Alternatively, `tls=true` can be specified in the [MongoDB Connection String]() passed to the initializer:
+```swift
+let client = try MongoClient("mongodb://example.com/?tls=true", using: elg)
+```
+**Note:** Specifying any `tls` prefixed option in the connection string or `ClientOptions` will cause MongoSwift to attempt to connect using TLS.
```

Comment:
nit: `tls`-prefixed, also this could maybe be changed to use the "required" wording as well